src/core/mac/mac_frame.cpp

```diff
@@ -1170,11 +1169,9 @@ otError TxFrame::GenerateEnhAck(const RxFrame &aFrame, bool aIsFramePending, con
     if (aFrame.GetSecurityEnabled())
     {
         SuccessOrExit(error = aFrame.GetSecurityControlField(securityControlField));
-        SuccessOrExit(error = aFrame.GetFrameCounter(frameCounter));
         SuccessOrExit(error = aFrame.GetKeyId(keyId));
 
         SetSecurityControlField(securityControlField);
-        SetFrameCounter(frameCounter);
         SetKeyId(keyId);
     }
 
```

Comment:
Could handle security here if `aIeLength == 0`

Reply:
The methods here are independent of any openthread instance object. It seems that we cannot access `SubMac` and process security here.